### ISSUE TITLE
Imports of legacy modules with numeric constants do not get flagged as unused when the corresponding type is used in the code

### ISSUE THREAD

**Issue by GrigorenkoPV** [rust-lang/rust#124654]
### Code

```Rust
use std::usize;

fn main() {
    let x: usize = 1;
    println!("{}", x);
}
```

### Current output

```Shell
`
```

### Desired output

```Shell
warning: unused import: `std::usize`
 --> src/main.rs:1:5
  |
1 | use std::usize;
  |     ^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
```

### Rust Version

```Shell
$ rustc --version --verbose
rustc 1.79.0-beta.1 (6b544f5ff 2024-04-28)
binary: rustc
commit-hash: 6b544f5ff8d45221d61962651a5f5ab9fe535e16
commit-date: 2024-04-28
host: x86_64-unknown-linux-gnu
release: 1.79.0-beta.1
LLVM version: 18.1.4
```

**Comment by GrigorenkoPV** [comment]
Technically this is a regression between Rust 1.8 & 1.9
